.github/workflows/build.yaml

```diff
@@ -13,7 +13,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v4
         with:
-          go-version: 1.20.x
+          go-version: 1.22.2
```

Comment:
why specify this version instead of 1.22.x

Reply:
As of Go 1.21, toolchain versions [must use the 1.N.P syntax](https://go.dev/doc/toolchain#version).
go.mod does not match this syntax and there is no additional toolchain directive, which may cause some go commands to fail.

Reply:
https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file

see here if we can just using `1.22`, and it will choose the latest patch version of `1.22.x` automatically.

Reply:
By the way, why we need to upgrade the Go version from `1.20` to `1.22`?

Reply:
> By the way, why we need to upgrade the Go version from `1.20` to `1.22`?

advance with the times. 😆😆
Also as the language develops, some toolchains may report errors.

Reply:
> https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file
> 
> see here if we can just using `1.22`, and it will choose the latest patch version of `1.22.x` automatically.

got it, let me try.

Reply:
> https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file
> 
> see here if we can just using `1.22`, and it will choose the latest patch version of `1.22.x` automatically.

It may be ok in action but it fails when I set it up locally, can you try to set it up locally in your area? @Ink-33

Reply:
> advance with the times.

No problem of course. It will be better if reviewers could see the specific reasons to upgrade Go version. For example:

* performance improve
* bugfix
* need some new APIs
* .etc.

Anyway, people could learn from those reasons. And this is a good to have suggestion.

Reply:
> > https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file
> > 
> > see here if we can just using `1.22`, and it will choose the latest patch version of `1.22.x` automatically.
> 
> It may be ok in action but it fails when I set it up locally, can you try to set it up locally in your area? @Ink-33 

Do you modify `go.mod`? Try to keep using 1.n.p in `go.mod` file and use `1.22` for github action.